helix-term/src/ui/editor.rs

```diff
@@ -695,6 +762,340 @@ impl EditorView {
         );
     }
 
+    /// Render the sticky context
+    #[allow(clippy::too_many_arguments)]
+    pub fn render_sticky_context(
+        doc: &Document,
+        view: &View,
+        surface: &mut Surface,
+        context: &Option<Vec<StickyNode>>,
+        doc_annotations: &TextAnnotations,
+        line_decoration: &mut [Box<dyn LineDecoration + '_>],
+        translated_positions: &mut [TranslatedPosition],
+        theme: &Theme,
+    ) {
+        let Some(context) = context else {
+            return;
+        };
+
+        let text = doc.text().slice(..);
+        let viewport = view.inner_area(doc);
+
+        // backup (status line) shall always exist
+        let status_line_style = theme
+            .try_get("ui.statusline")
+            .expect("`ui.statusline` exists");
+
+        // define sticky context styles
+        let context_style = theme
+            .try_get("ui.sticky.context")
+            .unwrap_or(status_line_style);
+        let indicator_style = theme
+            .try_get("ui.sticky.indicator")
+            .unwrap_or(status_line_style);
+
+        let mut context_area = viewport;
+        context_area.height = 1;
+
+        for node in context {
+            surface.clear_with(context_area, context_style);
+
+            let mut new_offset = view.offset;
+            let mut line_context_area = context_area;
+
+            if let Some(indicator) = node.indicator.as_deref() {
+                // set the indicator
+                surface.set_stringn(
+                    line_context_area.x,
+                    line_context_area.y,
+                    indicator,
+                    indicator.len(),
+                    indicator_style,
+                );
+                continue;
+            }
+
+            // get the len of bytes of the text that will be written (the "definition" line)
+            let line = text.line(node.line);
+            let tab_count = line.chars().take_while(|c| *c == '\t').count();
+
+            let already_written =
+                (line.len_bytes() + tab_count.saturating_mul(doc.tab_width() - 1)) as u16;
+
+            let dots = "...";
+
+            let mut cleared_virtual_text_annotations = doc_annotations.clone();
+            cleared_virtual_text_annotations.clear_line_annotations();
+            cleared_virtual_text_annotations.clear_inline_annotations();
```

Comment:
I think you can just create a new `TextAnnotations` struct here instead of cleraing the doc annotations. You don't want to ever carry those over IMO